Public/Src/Cache/ContentStore/Library/Sessions/ServiceClientContentSession.cs

```diff
@@ -38,65 +38,47 @@ public class ServiceClientContentSession : ReadOnlyServiceClientContentSession,
         }
 
         /// <inheritdoc />
-        protected override Task<PutResult> PutStreamCoreAsync(
+        protected override async Task<PutResult> PutStreamCoreAsync(

```

Comment:
What was reason for async/await-ifying this? As a rule, if all pieces of code are fast, it's better to not use async/await, because async/await incurs the overhead of allocating a state-carrying object and waiting for the scheduler to schedule you.

Reply:
We need to dispose of the stream in the case we need to create a temporary file. The alternative is to use ContinueWith and return a Task as we were doing before. The await syntax seems much more explicit about intent.